interfaces/builtin/microstack_support.go

```diff
@@ -0,0 +1,267 @@
+
+# Allow creating dm-* devices, /dev/<vg-name> directories, /dev/mapper directory and symlinks under it.
+# Allow issuing ioctls to the Device Mapper for LVM tools via /dev/mapper/control.
+/dev/mapper/control rw,
+# Besides symlinks for LVs prefixed with a VG name this is also needed for DM devices created with
+# dm-crypt and other DM modules.
+/dev/mapper/{,**} rw,
+# Allow device mapper devices to be accessed.
+/dev/dm-* rw,
+/dev/microstack-*/{,**} rw,
+# Allow bcache devices to be accessed since DM devices may be set up on top of those.
+/dev/bcache[0-9]{,[0-9],[0-9][0-9]} rw,                   # bcache (up to 1000 devices)
+
+# Allow access to loop devices and loop-control to be able to associate a file with a loop device
+# for the purpose of using a file-backed LVM setup.
+/dev/loop-control rw,
+/dev/loop[0-9]* rw,
+
+# Description: this policy intentionally allows Microstack services to configure AppArmor
+# as libvirt generates AppArmor profiles for the utility processes it spawns.
+/sys/kernel/security/apparmor/{,**} r,
+/sys/kernel/security/apparmor/.remove w,
+/sys/kernel/security/apparmor/.replace w,
+
+# Used by libvirt to work with IOMMU.
+/sys/kernel/iommu_groups/{,**} r,
+/sys/bus/pci/devices/**/iommu_group/** r,
+
+# Used by libvirt's QEMU driver state initialization code path.
+# The path used is hard-coded in libvirt to <huge-page-mnt-dir>/libvirt/qemu.
+/dev/hugepages/libvirt/ rw,
+/dev/hugepages/libvirt/** mrwklix,
+
+# Used by QEMU to get the maximum number of memory regions allowed in the vhost kernel module.
+/sys/module/vhost/parameters/max_mem_regions r,
+
+# Used by libvirt (cgroup-related):
+/sys/fs/cgroup/unified/cgroup.controllers r,
+/sys/fs/cgroup/cpuset/cpuset.cpus r,
+
+# Non-systemd layout: https://libvirt.org/cgroups.html#currentLayoutGeneric
+/sys/fs/cgroup/*/ r,
+/sys/fs/cgroup/*/machine/ rw,
+/sys/fs/cgroup/*/machine/** rw,
+
+# systemd-layout: https://libvirt.org/cgroups.html#systemdLayout
+/sys/fs/cgroup/*/machine.slice/machine-qemu*/{,**} rw,
+
+@{PROC}/[0-9]*/cgroup r,
+@{PROC}/cgroups r,
+
+# Used by libvirt.
+@{PROC}/filesystems r,
+@{PROC}/mtrr w,
+@{PROC}/@{pids}/environ r,
+@{PROC}/@{pids}/sched r,
+
+@{PROC}/*/status r,
+
+@{PROC}/sys/fs/nr_open r,
+
+# Libvirt needs access to the PCI config space in order to be able to reset devices.
+/sys/devices/pci*/**/config rw,
+
+# Spice
+owner /{dev,run}/shm/spice.* rw,
+
+# Used by libvirt to create lock files for /dev/pts/<num> devices
+# when handling virsh console access requests.
+/run/lock/ r,
+/run/lock/LCK.._pts_* rwk,
+
+# Used by LVM tools.
+/run/lock/lvm/ rw,
+/run/lock/lvm/** rwk,
+# Files like /run/lvm/pvs_online, /run/lvm/vgs_online, /run/lvm/hints
+/run/lvm/ rw,
+/run/lvm/** rwlk,
+/run/dmeventd-client rwlk,
+/run/dmeventd-server rwlk,
+
+# Used by targetcli tools to work with LIO.
+/sys/kernel/config/target/ rw,
+/sys/kernel/config/target/** rw,
+
+# Used by targetcli.
+/run/targetcli.lock rwlk,
+/var/run/targetcli.lock rwlk,
```

Comment:
this should be `/{var/,}run/...` shouldn't it? since /var/run is a symlink to /run?

Reply:
Ack, ty,